pages/visitors/[id]/name.js

```diff
@@ -0,0 +1,93 @@
+import React, { useCallback, useState } from "react";
+import { useRouter } from "next/router";
+import Button from "../../../src/components/Button";
+import FormGroup from "../../../src/components/FormGroup";
+import { GridRow, GridColumn } from "../../../src/components/Grid";
+import Hint from "../../../src/components/Hint";
+import Input from "../../../src/components/Input";
+import Layout from "../../../src/components/Layout";
+import LabelHeader from "../../../src/components/LabelHeader";
+import Text from "../../../src/components/Text";
+import BackLink from "../../../src/components/BackLink";
+import ErrorSummary from "../../../src/components/ErrorSummary";
+
+const Name = () => {
+  const router = useRouter();
+  const nameError = "Enter your name";
+  const [name, setName] = useState("");
+  const [errors, setErrors] = useState([]);
+
+  const hasError = (field) =>
+    errors.find((error) => error.id === `${field}-error`);
+
+  const onSubmit = useCallback(async (event) => {
+    event.preventDefault();
+    const errors = [];
+
+    if (!name) {
+      errors.push({
+        id: "name-error",
+        message: nameError,
+      });
+    }
+
+    setErrors(errors);
+
+    if (errors.length === 0) {
+      router.push(`/visits/${router.query.id}?name=${name}`);
+    }
+  });
+
+  return (
+    <Layout
+      title="What is your name? - Attend a virtual visit"
+      hasErrors={errors.length != 0}
+    >
+      <GridRow>
+        <GridColumn width="two-thirds">
+          <BackLink href={`/visitors/${router.query.id}/start`}>
```

Comment:
Just a question after reading the position of backlinks here [(https://service-manual.nhs.uk/design-system/components/back-link)] it might be more consistent to put it at the bottom?

Reply:
Yes, good spot!

We've updated `Layout` to take in a back link so we can easily change this in the future if necessary.